test/unit/project-management/project-management.spec.ts

```diff
@@ -380,19 +388,174 @@ describe('ProjectManagement', () => {
+      const partialApiResponse = {};
+
+      const stub = sinon
+          .stub(ProjectManagementRequestHandler.prototype, 'listAppMetadata')
+          .returns(Promise.resolve(partialApiResponse));
+      stubs.push(stub);
+      return projectManagement.listAppMetadata()
+          .should.eventually.deep.equal([]);
+    });
+
+    it('should throw when API response has non-array "apps" field', () => {
+      const partialApiResponse = { apps: 'none' };
+
+      const stub = sinon
+          .stub(ProjectManagementRequestHandler.prototype, 'listAppMetadata')
+          .returns(Promise.resolve(partialApiResponse));
+      stubs.push(stub);
+      return projectManagement.listAppMetadata()
+          .should.eventually.be.rejected
+          .and.have.property(
+              'message',
+              '"apps" field must be present in the listAppMetadata() response data. Response data: '
+                  + JSON.stringify(partialApiResponse, null, 2));
+    });
+
+    it('should throw with API response missing "apps[].appId" field', () => {
+      const partialApiResponse = {
+        apps: [{}],
+      };
+
+      const stub = sinon
+          .stub(ProjectManagementRequestHandler.prototype, 'listAppMetadata')
+          .returns(Promise.resolve(partialApiResponse));
+      stubs.push(stub);
+      return projectManagement.listAppMetadata()
+          .should.eventually.be.rejected
+          .and.have.property(
+              'message',
+              '"apps[].appId" field must be present in the listAppMetadata() response data. '
+                  + `Response data: ${JSON.stringify(partialApiResponse, null, 2)}`);
+    });
+
+    it('should throw with API response missing "apps[].platform" field', () => {
+      const missingPlatformApiResponse = {
+        apps: [{
+          appId: APP_ID,
+        }],
+      };
+
+      const stub = sinon
+          .stub(ProjectManagementRequestHandler.prototype, 'listAppMetadata')
+          .returns(Promise.resolve(missingPlatformApiResponse));
+      stubs.push(stub);
+      return projectManagement.listAppMetadata()
+          .should.eventually.be.rejected
+          .and.have.property(
+              'message',
+              '"apps[].platform" field must be present in the listAppMetadata() response data. '
+                  + `Response data: ${JSON.stringify(missingPlatformApiResponse, null, 2)}`);
+    });
+
+    it('should resolve with list of apps metadata on success', () => {
+      const expectedAppMetadata: AppMetadata[] = [
+        {
+          appId: VALID_LIST_APP_METADATA_API_RESPONSE.apps[0].appId,
+          displayName: VALID_LIST_APP_METADATA_API_RESPONSE.apps[0].displayName,
+          platform: AppPlatform.ANDROID,
+          projectId: mocks.projectId,
+          resourceName: RESOURCE_NAME_ANDROID,
+        },
+        {
+          appId: VALID_LIST_APP_METADATA_API_RESPONSE.apps[1].appId,
+          displayName: VALID_LIST_APP_METADATA_API_RESPONSE.apps[1].displayName,
```

Comment:
Can you update this test to make exactly 1 of the apps have no display name, to test your latest change?